### PR TITLE
Also test on PyPy 3.7

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

See https://github.com/PyCQA/pylint/pull/5849#discussion_r816766177.

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
I'm ok with fixing the issue with the line/col either in 2.13 or 2.14 :)

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -476,7 +476,7 @@ jobs:
         with:
           path: venv
           key:
-            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{
+            ${{ runner.os }}-${{ matrix.python-version }}-${{
```
Comment (cdce8p) [inline .github/workflows/ci.yaml:479]: This shouldn't be necessary. The `Set up Python` version will output `${{ matrix.python-version }}`. As far as I can tell, the tests didn't crash.

Comment (DanielNoord) [reply .github/workflows/ci.yaml:479]: Hmm, this doesn't work as well.

Please see the version of `pytest` in the `pypy-3.7` run. That's the `3.6` version. You can see they are using the same cache in the `Restore Python virtual environment` job. The `key` is the same.

However, my fix also doesn't work. I guess something weird is going on with `python-version` and `PyPy`? For the `cpython` jobs the keys do work..

**Comment by DanielNoord** [comment]
There is an issue with the current caching on `PyPy`. For some reason `steps.python.outputs.python-version` is empty on the `PyPy` jobs. I'm not sure why.
You can see in the first commit checks that `pypy-3.7` actually uses `3.6` for `pytest`. They share the same cache. For `3.6` and `3.7` this isn't a problem, but it will be when we add `3.8` and it was also the causes of the issues I had in #5849.

I could do a separate PR for the fix, but I think it should be fine to squash here, or to add two commits and rebase here.

-------

For the future: we should remove `3.6` when we officially drop support of `3.6`. After that we should look into testing `PyPy` on `3.8` as well and seeing what that requires from us.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cdce8p) [reply .github/workflows/ci.yaml:479]: This pretty much looks like a bug in the `setup-python` actions. If I had to guess a result of https://github.com/actions/setup-python/pull/342.

Comment (cdce8p) [reply .github/workflows/ci.yaml:479]: I've included the fix in #5868
Not sure when the bug will be addressed. We can always revert it, once a new version is released.

After #5868 is done, we should be able to move forward here.

**Review by cdce8p — APPROVED** [review]
I've rebased the PR to only include the addition of `pypy-3.7`.
Don't know if you wanted to do anything else, so will  leave it as draft for now. Feel free to merge it though if it's finished.

**Comment by DanielNoord** [comment]
Thanks @cdce8p, I'm going to merge this!

**Pull request merged**